File: gibbs.js
var jStat = require('jStat').jStat;

function gibbs(N, thin) {
    var x = 0.0;
    var y = 0.0;
    console.log("Iter  x  y");
    for (i=0;i<N;i++) {
        for (j=0;j<thin;j++) {
            x=jStat.gamma.sample(3, 1.0/(y*y+4));
            y=jStat.normal.sample(1.0/(x+1), 1.0/Math.sqrt(2*x+2));
        }
        console.log(i, x, y);
    }
}

gibbs(50000, 1000);

File: gibbs.jl
using Distributions

function gibbs(N, thin)
    x = 0
    y = 0
    println("Iter  x  y")
    i = 0
    for i = 1:N
        for j = 0:thin
            x = rand(Gamma(3, 1.0/(y*y+4)))
            y = rand(Normal(1.0/(x+1), 1/sqrt(2*x+2)))
        end
        println(i, ' ', x, ' ', y)
    end
end

gibbs(50000, 1000)
